aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -326,6 +341,113 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    public entry fun rotate_authentication_key_with_rotation_capability(
+        delegate_signer: &signer,
+        rotation_cap_offerer_address: address,
+        new_scheme: u8,
+        new_public_key_bytes: vector<u8>,
+        cap_update_table: vector<u8>
+    ) acquires Account, OriginatingAddress {
+        assert!(exists_at(rotation_cap_offerer_address), error::not_found(EOFFERER_ADDRESS_DOES_NOT_EXIST));
+
+        // check that there exists a rotation capability offer at the offerer's account resource for the delegate
+        let offerer_account_resource = borrow_global<Account>(rotation_cap_offerer_address);
+        let delegate_address = signer::address_of(delegate_signer);
+        assert!(option::contains(&offerer_account_resource.rotation_capability_offer.for, &delegate_address), error::not_found(ENO_SUCH_ROTATION_CAPABILITY_OFFER));
+
+        let delegate_account_resource = borrow_global<Account>(delegate_address);
+        let curr_auth_key = from_bcs::to_address(offerer_account_resource.authentication_key);
+        let challenge = RotationProofChallenge {
```

Comment:
Is this challenge just to ensure the delegate_signer is in possession of the auth key?

Reply:
I don't quite follow why this is a concern at all?

table::add(address_map, new_address, rotation_cap_offerer_address);

Because the auth key is updated to allow new_address to control the offerer's account, updating OriginatingAddress signifies the same thing. I don't see why this is malicious?

Reply:
Here's the attack.

Alice has rotated her account `addr_a` to `new_addr_a`. As a result, the following entry is created, to help Alice when recovering her wallet: `OriginatingAddress[new_addr_a] -> addr_a`

Alice has had bad day: her laptop blew up and she needs to reset her account on a new one. Fortunately, she still has her secret key `new_sk_a` associated with her new address `new_addr_a`.

But Bob likes to mess with Alice.

Bob will create an account `addr_b` and maliciously rotates it to Alice's new address `new_addr_a`. Since we are no longer checking a PoK, Bob can easily do this.

Now, the table will be updated to make Alice's new address point to Bob's address:  `OriginatingAddress[new_addr_a] -> addr_b`.

When Alice recovers her account, her wallet will display the attacker's address (Bob's) `addr_b` as her address.

Now Alice will give `addr_b` to everyone to pay her, but the money will go to Bob.

Reply:
sounds good - will add more comments, thanks for the great example : )